org.mwc.debrief.core/src/org/mwc/debrief/core/ContextOperations/GeneratePasteRepClipboard.java

```diff
@@ -0,0 +1,156 @@
+/*
+ *    Debrief - the Open Source Maritime Analysis Application
+ *    http://debrief.info
+ *
+ *    (C) 2000-2018, Deep Blue C Technology Ltd
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the Eclipse Public License v1.0
+ *    (http://www.eclipse.org/legal/epl-v10.html)
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ */
+package org.mwc.debrief.core.ContextOperations;
+
+import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IMenuManager;
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.mwc.cmap.core.CorePlugin;
+import org.mwc.cmap.core.operations.CMAPOperation;
+import org.mwc.cmap.core.property_support.RightClickSupport.RightClickContextItemGenerator;
+
+import Debrief.ReaderWriter.Replay.ImportReplay;
+import MWC.GUI.Editable;
+import MWC.GUI.Layer;
+import MWC.GUI.Layers;
+
+/**
+ * Generates a paste REP from clipboard action.
+ * @author Ayesha <ayesha.ma@gmail.com>
+ *
+ */
+public class GeneratePasteRepClipboard implements RightClickContextItemGenerator
+{
+
+  /* (non-Javadoc)
+   * @see org.mwc.cmap.core.property_support.RightClickSupport.RightClickContextItemGenerator#generate(org.eclipse.jface.action.IMenuManager, MWC.GUI.Layers, MWC.GUI.Layer[], MWC.GUI.Editable[])
+   */
+  @Override
+  public void generate(IMenuManager parent, Layers theLayers,
+      Layer[] parentLayers, Editable[] subjects)
+  {
+    //see if there is nothing selected 
+    if(subjects.length==0) {

```

Comment:
Oh dear, Ian's being picky again.  I think your eclipse is configured with the wrong braces format.  We have braces on a newline.

Here are steps to import the project styles:
https://github.com/debrief/debrief/wiki/On-boarding#project-settings